runtime/doc/lua.txt

```diff
@@ -2230,6 +2230,12 @@ vim.tbl_count({t})                                           *vim.tbl_count()*
 vim.tbl_deep_extend({behavior}, {...})                 *vim.tbl_deep_extend()*
     Merges recursively two or more tables.
 
+    Only values that are empty tables or tables that are not |lua-list|s
+    (indexed by consecutive integers starting from 1) are merged recursively.
+    This is useful for merging nested tables like default and user
+    configurations where lists should be treated as literals (i.e., are
```

Comment:
```suggestion
    configurations where nested lists should be treated as literals (i.e., are
```

Reply:
Does it make a difference (honestly asking)?

Reply:
Are top-level lists merged?

Anyway, I think the "nested tables" on the previous line perhaps is the important part, so my suggestion can be ignored.

Reply:
> Are top-level lists merged?

No, as @echasnovski points out above. Only lists that are _values_ are special-cased.

Reply:
(Always happy about suggestions for improvement; I'd just like to understand them to learn from them.)